aea/agent.py

```diff
@@ -192,10 +174,19 @@ def tick(self) -> int:  # pragma: nocover
         """
         return self._tick
 
+    def handle_envelope(self, envelope: Envelope) -> None:
+        """
+        Handle an envelope.
+
+        :param envelope: the envelope to handle.
+        :return: None
+        """
+        raise NotImplementedError
+
     @property
-    def timeout(self) -> float:
-        """Get the time in (fractions of) seconds to time out an agent between act and react."""
-        return self._timeout
+    def period(self) -> float:
+        """Get a period to call act."""
+        return self._period
 
     @property
     def runtime(self) -> BaseRuntime:
```

Comment:
Runtime should be an abstractproperty

Reply:
Why? it's a concrete property of the agent implementation. it should state in Agent class, not AbstractAgent.